Cython/Compiler/ModuleNode.py

```diff
@@ -876,14 +876,17 @@ def generate_module_preamble(self, env, options, cimported_modules, metadata, co
         if has_np_pythran(env):
             env.use_utility_code(UtilityCode.load_cached("PythranConversion", "CppSupport.cpp"))
 
-    def generate_extern_c_macro_definition(self, code):
+    def generate_extern_c_macro_definition(self, code, env):
         name = Naming.extern_c_macro
         code.putln("#ifndef %s" % name)
-        code.putln("  #ifdef __cplusplus")
-        code.putln('    #define %s extern "C"' % name)
-        code.putln("  #else")
-        code.putln("    #define %s extern" % name)
-        code.putln("  #endif")
+        if env.is_cpp():
```

Comment:
If this is the only functionality that we need from env, then I'd rather add an argument `is_cpp=False` than passing  all of`env` into this method.

Reply:
Agreed.